### PR TITLE
Feature/update gutenberg plugin

### PR DESCRIPTION
**Pull request by jomurgel**
- Fixes setting post status in the Gutenberg context.
- Adds noindex/norobots meta if is `archiveless`.
- Updates readme.

### CODE REVIEW

File: class-archiveless.php [hunk 1]
```diff
@@ -264,5 +318,18 @@ public function inline_locale_data( string $to_handle ) {
...
 		);
 	}
+
+	/**
+	 * Return robots meta if archiveless.
+	 */
+	public function no_index() {
+		global $post;
+
+		var_dump( $post );
```
Comment (mboynes) [inline class-archiveless.php]: debug code
```suggestion
```

Comment (jomurgel) [reply class-archiveless.php]: oops! Thanks!

File: class-archiveless.php [hunk 2]
```diff
@@ -174,6 +182,52 @@ public function wp_insert_post_data( $data, $postarr ) {
 		return $data;
 	}
 
+	/**
+	 * Set the custom post status when post data is being inserted.
+	 *
+	 * WordPress, unfortunately, doesn't provide a great way to _manage_ custom
+	 * post statuses. While we can register and use them just fine, there are
+	 * areas of the Admin where statuses are hard-coded. This method is part of
```
Comment (pattiereaves) [inline class-archiveless.php]:
```suggestion
	 * areas of the admin where statuses are hard-coded. This method is part of
```

File: class-archiveless.php [hunk 3]
```diff
@@ -174,6 +182,52 @@ public function wp_insert_post_data( $data, $postarr ) {
...
+	 * this plugin's trickery to provide a seamless integration.
+	 *
+	 * @param  object $prepared_post Post data. Arrays are expected to be escaped, objects are not. Default array.
+	 * @return array $data, potentially with a new status.
```
Comment (pattiereaves) [inline class-archiveless.php]:
```suggestion
	 * @return array $data Post data, potentially with a new status.
```

**Pull request merged**